src/challenges/challengeTwo/IntegrationComponent/IntegrationComponent.test.tsx

```diff
@@ -1,8 +1,88 @@
-// import React from "react";
-// import { render, screen } from "@testing-library/react";
-// import userEvent from "@testing-library/user-event";
-// import { IntegrationComponent } from "./IntegrationComponent";
-
-// describe("IntegrationComponent", () => {
-//   it("renders...", () => {});
-// });
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import axios from "axios";
+import { IntegrationComponent } from "./IntegrationComponent";
+
+jest.mock("axios");
+const mockWeatherData = {
+  temperature: "18",
+  weather: "cool",
+};
+describe("IntegrationComponent", () => {
+  const renderComponent = (additionalProps?: any) => {
+    render(<IntegrationComponent {...additionalProps} />);
+  };
+
+  it("render weather data", async () => {
+    (axios.get as jest.MockedFunction<typeof axios.get>).mockResolvedValueOnce({
+      data: mockWeatherData,
+    });
+    renderComponent();
+    expect(await screen.findByText(/London Weather/i)).toBeInTheDocument();
+    expect(await screen.findByText(/18/i)).toBeInTheDocument();
+    expect(await screen.findByText(/cool/i)).toBeInTheDocument();
+  });
+
+  it("opens the form modal on button click", async () => {
+    renderComponent();
+    const openFormButton = screen.getByRole("button");
+    await userEvent.click(openFormButton);
+    expect(screen.getByText("Form Modal")).toBeInTheDocument();
+  });
+
+  it("closes the modal when clicking on the cross icon", async () => {
+    renderComponent();
+    const openFormButton = screen.getByText("Open Form");
+    await userEvent.click(openFormButton);
+
+    const crossIcon = screen.getByTestId("modal-close-icon");
+    await userEvent.click(crossIcon);
+
+    expect(screen.queryByText("Form Modal")).not.toBeInTheDocument();
+  });
+
+  it("render form field", async () => {
+    renderComponent();
+    const openFormButton = screen.getByText("Open Form");
+    await userEvent.click(openFormButton);
+    expect(screen.getByLabelText("Name")).toBeInTheDocument();
+    expect(screen.getByLabelText("Email")).toBeInTheDocument();
+  });
+
+  it("updates name field values on input change", async () => {
+    renderComponent();
+    const openFormButton = screen.getByText("Open Form");
+    await userEvent.click(openFormButton);
+    const inputElement = screen.getByLabelText("Name");
+    await userEvent.type(inputElement, "Lovepreet");
+    expect(inputElement).toHaveValue("Lovepreet");
+  });
+
+  it("updates email field values on input change", async () => {
```

Comment:
A great test file but I think you over-tested in this integration test file. These tests are identical to your model tests so you're testing twice. 

So really all you needed was four tests:

1. Weather data displays correctly
2. Modal opens when clicking the button
3. Modal submits correctly and shows the correct success message
4. Modal closes after submitting